var/spack/repos/builtin/packages/libbeagle/package.py

```diff
@@ -41,7 +41,12 @@ class Libbeagle(AutotoolsPackage):
 
     depends_on('subversion', type='build')
     depends_on('pkgconfig', type='build')
+    depends_on('java', type='build')
 
     def url_for_version(self, version):
         url = "https://github.com/beagle-dev/beagle-lib/archive/beagle_release_{0}.tar.gz"
         return url.format(version.underscored)
+
+    def setup_dependent_environment(self, spack_env, run_env, dependent_spec):
+        prefix = self.prefix
+        spack_env.prepend_path('LD_LIBRARY_PATH', prefix.lib)
```

Comment:
This should not be necessary. This implies that the package you are trying to build that depends on `libbeagle` is not using Spack's compiler wrappers.

Reply:
beast1 is a java program with a wrapper script

The wrapper script is looking for the library in:
```java.library.path="$BEAST_LIB:/usr/local/lib:$LD_LIBRARY_PATH"```

I could modify the beast1 package to add self.spec['libbeagle'].libs to the java.library.path instead if that makes more sense

Reply:
Hmm, that makes sense. Could we instead set `BEAST_LIB`? Spack unsets `LD_LIBRARY_PATH` by default, although maybe it sets `spack_env` afterwards. This will have the added benefit that if someone has `BEAST_LIB` already set via a module file, it won't pick up the wrong installation.

Reply:
Sure, that works - updated